src/sentry/notifications/utils/participants.py

```diff
@@ -171,6 +171,16 @@ def get_owners(project: Project, event: Event | None = None) -> Iterable[Team |
         outcome = "match"
         recipients = ActorTuple.resolve_many(owners)
 
+    ownership = ProjectOwnership.objects.filter(project=project).first()
```

Comment:
I think we we can skip the DB call and flag check if `len(recipients) <= 1`.

Reply:
i dont know if theres a way around it, unfortunately we need the `ownership.fallthrough` in the if statement after this line

Reply:
```py
if len(recipients) > 1:
    ownership = ProjectOwnership.objects.filter(project=project).first()
    # Used to suppress extra notifications to all project members, only notify the would-be auto-assignee
    if (
        ownership
        and not ownership.fallthrough
        and not features.has("organizations:notification-all-recipients", project.organization)
    ):
        return list(recipients)[-1:]
```

Reply:
Let's use `ProjectOwnership.get_ownership_cached`. Alternatively we could return this value from `get_owners`

Reply:
@mgaeta  ahhh, i misunderstood what you mean, implemented that
@wedamija  didn't notice that method, thanks